### PR TITLE
[Feature] Support to_tera_date/to_tera_timestamp

### PR DESCRIPTION
**Pull request by LiShuMing**
Fixes #issue

Support `to_tera_date `/`to_tera_timestamp` function to be compatible with trino : https://trino.io/docs/current/functions/list-by-topic.html?highlight=teradata#teradata

## What type of PR is this:
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/exprs/time_functions.cpp [hunk 1]
```diff
@@ -646,6 +646,134 @@ DEFINE_UNARY_FN_WITH_IMPL(to_dateImpl, v) {
 }
 DEFINE_TIME_UNARY_FN(to_date, TYPE_DATETIME, TYPE_DATE);
 
+struct TeradataFormatState {
+    std::unique_ptr<TeradataFormat> formatter;
+};
+
+// to_date_v2
+Status TimeFunctions::to_date_v2_prepare(FunctionContext* context, FunctionContext::FunctionStateScope scope) {
```
Comment (Seaven) [inline be/src/exprs/time_functions.cpp:654]: `to_teradata` is better?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/exprs/time_functions.cpp (hunk 1 above)
Comment (LiShuMing) [reply be/src/exprs/time_functions.cpp]: fixed.

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 194 / 241 (80.50%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/runtime/datetime_value.cpp|126|161|78.26%|[1954, 1964, 1970, 1980, 1988, 1996, 2007, 2010, 2016, 2020, 2030, 2033, 2039, 2043, 2058, 2066, 2077, 2084, 2090, 2097, 2108, 2109, 2110, 2112, 2113, 2115, 2116, 2117, 2118, 2119, 2127, 2134, 2138, 2139, 2154]|
|:large_blue_circle:|src/exprs/time_functions.cpp|62|74|83.78%|[656, 660, 668, 690, 695, 701, 717, 721, 729, 753, 758, 764]|
|:large_blue_circle:|src/runtime/datetime_value.h|6|6|100.00%|[]|

**Review by murphyatwork — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 194 / 241 (80.50%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/runtime/datetime_value.cpp|126|161|78.26%|[1954, 1964, 1970, 1980, 1988, 1996, 2007, 2010, 2016, 2020, 2030, 2033, 2039, 2043, 2058, 2066, 2077, 2084, 2090, 2097, 2108, 2109, 2110, 2112, 2113, 2115, 2116, 2117, 2118, 2119, 2127, 2134, 2138, 2139, 2154]|
|:large_blue_circle:|src/exprs/time_functions.cpp|62|74|83.78%|[656, 660, 668, 690, 695, 701, 717, 721, 729, 753, 758, 764]|
|:large_blue_circle:|src/runtime/datetime_value.h|6|6|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 194 / 241 (80.50%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/runtime/datetime_value.cpp|126|161|78.26%|[1954, 1964, 1970, 1980, 1988, 1996, 2007, 2010, 2016, 2020, 2030, 2033, 2039, 2043, 2058, 2066, 2077, 2084, 2090, 2097, 2108, 2109, 2110, 2112, 2113, 2115, 2116, 2117, 2118, 2119, 2127, 2134, 2138, 2139, 2154]|
|:large_blue_circle:|src/exprs/time_functions.cpp|62|74|83.78%|[656, 660, 668, 690, 695, 701, 717, 721, 729, 753, 758, 764]|
|:large_blue_circle:|src/runtime/datetime_value.h|6|6|100.00%|[]|

**Comment by LiShuMing** [comment]
@mergfyio backport branch-3.2

**Comment by LiShuMing** [comment]
@mergifyio backport branch-3.2

**Pull request merged**